src/platforms/ios/ioscontroller.swift

```diff
@@ -116,52 +135,31 @@ public class IOSControllerImpl : NSObject {
 
-            let keyData = PublicKey(base64Key: serverPublicKey)!
-            let dnsServerIP = IPv4Address(dnsServer)
-            let ipv6GatewayIP = IPv6Address(serverIpv6Gateway)
-
-            var peerConfiguration = PeerConfiguration(publicKey: keyData)
-            peerConfiguration.endpoint = Endpoint(from: serverIpv4AddrIn + ":\(serverPort )")
-            peerConfiguration.allowedIPs = []
-
-            allowedIPAddressRanges.forEach {
-                if (!$0.isIpv6) {
-                    peerConfiguration.allowedIPs.append(IPAddressRange(address: IPv4Address($0.address as String)!, networkPrefixLength: $0.networkPrefixLength))
-                } else {
-                    peerConfiguration.allowedIPs.append(IPAddressRange(address: IPv6Address($0.address as String)!, networkPrefixLength: $0.networkPrefixLength))
-                }
-            }
-
-            var peerConfigurations: [PeerConfiguration] = []
-            peerConfigurations.append(peerConfiguration)
-
-            var interface = InterfaceConfiguration(privateKey: self.privateKey!)
-
-            if let ipv4Address = IPAddressRange(from: deviceIpv4Address!),
-               let ipv6Address = IPAddressRange(from: deviceIpv6Address!) {
-                interface.addresses = [ipv4Address, ipv6Address]
+            let config = createConfig(for: serverData, allowedIPAddressRanges: allowedIPAddressRanges)
+            let fallbackConfig: TunnelConfiguration?
+            if let fallbackServer = fallbackServer {
+                fallbackConfig = createConfig(for: fallbackServer, allowedIPAddressRanges: allowedIPAddressRanges)
+            } else {
+                fallbackConfig = nil
             }
-            interface.dns = [DNSServer(address: dnsServerIP!), DNSServer(address: ipv6GatewayIP!)]
 
-            let config = TunnelConfiguration(name: VPN_NAME, interface: interface, peers: peerConfigurations)
-
-            return self.configureTunnel(config: config, reason: reason, serverName: serverIpv4AddrIn + ":\(serverPort )", excludeLocalNetworks: excludeLocalNetworks, gleanDebugTag: gleanDebugTag, isSuperDooperFeatureActive: isSuperDooperFeatureActive, installationId: installationId, disconnectOnErrorCallback: disconnectOnErrorCallback, onboardingCompletedCallback: onboardingCompletedCallback, vpnConfigPermissionResponseCallback: vpnConfigPermissionResponseCallback)
+            return self.configureTunnel(config: config, reason: reason, serverName: serverData.ipv4AddrIn + ":\(serverData.port )", fallbackConfig: fallbackConfig, excludeLocalNetworks: excludeLocalNetworks, gleanDebugTag: gleanDebugTag, isSuperDooperFeatureActive: isSuperDooperFeatureActive, installationId: installationId, disconnectOnErrorCallback: disconnectOnErrorCallback, onboardingCompletedCallback: onboardingCompletedCallback, vpnConfigPermissionResponseCallback: vpnConfigPermissionResponseCallback)
```

Comment:
nit: Consider adding a function to the VPNServerData struct that returns serverName and skip the extra argument?

Reply:
Ah, duh 🤦‍♀️ That's right.